units/en/unit2/continue-client.mdx

```diff
@@ -0,0 +1,134 @@
+# Using MCP with Local and Open Source Models
+
+In this section, we'll connect MCP with local and open-source models using
+Continue, a tool for building AI coding assistants that works with local tools
+like Ollama.
+
+## Setup Continue
+
+You can install Continue from the VS Code marketplace.
+
+<Tip>
+*Continue also has an extension for JetBrains.*
+</Tip>
+
+### VS Code extension
+
+1. Click `Install` on the [Continue extension page in the Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=Continue.continue)
+2. This will open the Continue extension page in VS Code, where you will need to click `Install` again
+3. The Continue logo will appear on the left sidebar. For a better experience, move Continue to the right sidebar
+
+![sidebar vs code demo](https://docs.continue.dev/assets/images/move-to-right-sidebar-b2d315296198e41046fc174d8178f30a.gif)
+
+With Continue configured, we'll move on to setting up Ollama to pull local models.
+
+### Ollama local models
+
+Ollama is an open-source tool that allows users to run large language models (LLMs)
+locally on their own computers. To use Ollama, you can [install](https://ollama.com/download) it and
+download the model you want to run with the `ollama run` command.
+
+For example, you can download the [llama 3.1:8b](https://ollama.com/models/llama-3:1b) model with:
+
+```bash
+ollama pull llama3.1:8b
+```
+
+It is important that we use models that have tool calling as a built-in feature, i.e. Codestral Qwen and Llama 3.1x.
+
+1. Create a folder called `.continue/models` at the top level of your workspace
+2. Add a file called `llama-max.yaml` to this folder
+3. Write the following contents to `llama-max.yaml` and save
+
+```yaml
+name: Ollama Llama model
+version: 0.0.1
+schema: v1
+models:
+  - provider: ollama
+    model: llama3.1:8b
+    defaultCompletionOptions:
+      contextLength: 50000
```

Comment:
```suggestion
      contextLength: 128000
```

The max context length for this model is actually 128k: https://ollama.com/library/llama3.1:8b

Reply:
Thanks. I was putting a random number here until it worked 😅